domain/src/main/java/org/cryptomator/domain/PCloud.java

```diff
@@ -0,0 +1,140 @@
+package org.cryptomator.domain;
+
+import org.jetbrains.annotations.NotNull;
+
+public class PCloud implements Cloud {
+
+	private final Long id;
+	private final String accessToken;
+	private final String url;
+	private final String username;
+
+	private PCloud(Builder builder) {
+		this.id = builder.id;
+		this.accessToken = builder.accessToken;
+		this.url = builder.url;
+		this.username = builder.username;
+	}
+
+	public static Builder aPCloud() {
+		return new Builder();
+	}
+
+	public static Builder aCopyOf(PCloud pCloud) {
+		return new Builder() //
+				.withId(pCloud.id()) //
+				.withAccessToken(pCloud.accessToken()) //
+				.withUrl(pCloud.url()) //
+				.withUsername(pCloud.username());
+	}
+
+	@Override
+	public Long id() {
+		return id;
+	}
+
+	public String accessToken() {
+		return accessToken;
+	}
+
+	public String url() {
+		return url;
+	}
+
+	public String username() {
+		return username;
+	}
+
+	@Override
+	public CloudType type() {
+		return CloudType.PCLOUD;
+	}
+
+	@Override
+	public boolean configurationMatches(Cloud cloud) {
+		return cloud instanceof PCloud && configurationMatches((PCloud) cloud);
+	}
+
+	private boolean configurationMatches(PCloud cloud) {
+		return url.equals(cloud.url) && username.equals(cloud.username);
```

Comment:
As this isn't something that comes from the user but from the cloud, this value could maybe change later which would lead to a new cloud. IMO this should be just the unique identifier which is the `cloud.username` 
```suggestion
		return username.equals(cloud.username);
```

Reply:
Afaik, you can switch between the data centers:
![grafik](https://user-images.githubusercontent.com/4191545/112488960-4c463f80-8d7e-11eb-8ff0-983e0469b68c.png)

Reply:
Ah, yeah, I made some false assumptions.